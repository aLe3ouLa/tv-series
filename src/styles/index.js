import React from "react";

import { Reset, Typography } from "./defaults";
import { color } from "./variables/colors";
import { spacing } from "./variables/spacings";
import { breakpoint } from "./variables/breakpoints";

export const Defaults = () => (
  <>
    <Reset />
    <Typography />
  </>
);

export const theme = {
  color,
  spacing,
  breakpoint,
};
