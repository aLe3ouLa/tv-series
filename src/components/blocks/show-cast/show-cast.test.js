import React from "react";
import ShowCast from "./";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../styles";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Show Cast component", () => {
  const mockProps = {
    cast: [{ person: { name: "test" }, character: { name: "test1" } }],
  };
  afterEach(cleanup);

  it("should render shows cast component without crashing", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowCast {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("section for episodes exists a li for a cast", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowCast {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const li = getByTestId("cast-list-0");
    expect(li).toBeInTheDocument();
  });

  it("section for cast should render an element with values test - test1", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <ShowCast {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const cast = getByTestId("cast-list-0");
    expect(cast.textContent).toBe("test - test1");
  });
});
