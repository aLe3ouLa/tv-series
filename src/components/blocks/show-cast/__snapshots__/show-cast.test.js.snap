// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Cast component should render shows cast component without crashing 1`] = `
<div>
  <section
    class="sc-AxjAm jHXAgG"
  >
    <h2>
      Cast
    </h2>
    <ul>
      <li
        data-testid="cast-list-0"
      >
        test
         - 
        test1
      </li>
    </ul>
  </section>
</div>
`;
